### DISCUSSION TITLE
May be a fast simulation object creation and assignment support is good for high level usage.

### DISCUSSION THREAD

**Discussion by Readon** [SpinalHDL/SpinalHDL#1136] [Ideas]
In the design, we normally define classes to extend Bundle for signals with specified target.
If we want to create object in simulation for one transaction when assigning to those Bundle, we need to assign them one by one through ```#=```, as I know.
It could be more helpful if we can inherit data class from previously defined bundle class as below
```
//in the design
case class Cmd() extends Bundle {
     val aSignal = Bool()
}

// during the simulation preparation, create object for simulation but with same structure as Cmd.
       val cmdObj = createSimObjFrom(Cmd)

// when simulator is running
       io.cmd #= cmdObj
```
So that the assignment to bundle is available. 
More useful thing is that we can create all transactions before simulation is running.
Or even further, some abstraction can be adopted to objects like ```cmdObj```.

**Comment by Dolu1990** [comment]
I think spinal.lib.sim.SimData is kindof like what you meant.
It isn't used much, but basicaly, it can capture/set a Data into/from a internal hashmap and allow to access member in a "natural" way via the "extends Dynamic".
It handle everything as BigInt.

**Reply by Dolu1990** [reply]
https://github.com/SpinalHDL/SpinalHDL/blob/5d404341bd43ddbfa0247efad46206250c67f1b0/lib/src/main/scala/spinal/lib/sim/SimData.scala#L16

**Reply by Readon** [reply]
That's cool, I will try that.
If I want to initialize an array of SimData for simulation, can I use it as below?
```
    val objects = Seq(Cmd(True), Cmd(False), Cmd(False))
    val objDatas = objects.map(SimData.copy(_))
    dut.io.cmd #= objDatas.randomPick()
```
Just making sure that what I understand is what it was designed to do.

**Reply by Dolu1990** [reply]
objDatas.randomPick().write(dut.io.cmd)   i think, else all good.

There is no #= implemented, could be added, in additon of a xxx.toSimData
